Make TorManager a HostedService

Kukks opened the pull request:
Hosted Services run as the first thing after DI container is built. Also cleaned up small bits

Kukks committed:
Make TorManager a HostedService

Hosted Services run as the first thing after DI container is built. Also cleaned up small bits

DanGould commented on Chaincase.Common/Global.cs:
```diff
@@ -365,18 +356,15 @@ public async Task OnResuming()
-                    {
-                        await _torManager.StartAsync(false, DataDir);
-                    }
+                    await _torManager.StartAsync(ResumeCts.Token);
```
Because it's written in c we don't have full control over the tor process. The conditional was added from the reference implementation @ OnionBrowser. Why remove?

DanGould commented on Chaincase.Common/Global.cs:
```diff
@@ -428,14 +416,10 @@ public async Task OnSleeping()
-                        await _torManager.StopAsync();
-                        Logger.LogInfo($"{nameof(Global)}.OnSleeping():{nameof(_torManager)} is stopped.");
-                    }
+                    await _torManager.StopAsync(SleepCts.Token);
```
Same here, vestige of https://github.com/OnionBrowser/OnionBrowser/blob/0f835926946c2b6c084098417a58fa59ecc3c98a/OnionBrowser/AppDelegate.swift#L280-L287

I'm not sure we want to remove the conditional

DanGould commented on Chaincase.iOS/Services/iOSTorManager.cs:
```diff
@@ -255,31 +254,31 @@ public static async Task<bool> IsTorRunningAsync(EndPoint torSocks5EndPoint)
         }
 
-        public async Task StopAsync()
+        public override async Task StopAsyncCore(CancellationToken cancellationToken)
```
why pass `cancellationToken` here? looks unecessary

DanGould commented:
Is this stale? Why do we want it a HostedService @Kukks

Kukks committed:
Flip dependency tree for TorManager <> Global

Reasoning:
For the eventual death of Global
Question here:
How critical is it that we wait for tor to actually init/stop?
